eaf.el

```diff
@@ -113,14 +115,45 @@ A bookmark handler function is used as
 A new app can use this to configure extensions which should
 handled by it.")
 
+(defvar eaf-build-dir (file-name-directory (locate-library "eaf")))
+(defvar eaf-source-dir (file-name-directory (file-truename (concat eaf-build-dir "eaf.el"))))
+
+(defcustom eaf-apps-to-install nil
+  "List of applications to install"
+  :group 'eaf
+  :type '(set
+	  (const :tag "EAF Airshare" eaf-airshare)
+	  (const :tag "EAF Browser" eaf-browser)
+	  (const :tag "EAF Camera" eaf-camera)
+	  (const :tag "EAF Demo" eaf-demo)
+	  (const :tag "EAF File Browser" eaf-file-browser)
+	  (const :tag "EAF File Manager" eaf-file-manager)
+	  (const :tag "EAF File Sender" eaf-file-sender)
+	  (const :tag "EAF Image Viewer" eaf-image-viewer)
+	  (const :tag "EAF Jupyter" eaf-jupyter)
+	  (const :tag "EAF Markdown Previewer" eaf-markdown-previewer)
+	  (const :tag "EAF Mermaid" eaf-mermaid)
+	  (const :tag "EAF Mindmap" eaf-mindmap)
+	  (const :tag "EAF Music Player" eaf-music-player)
+	  (const :tag "EAF Org Previewer" eaf-org-previewer)
+	  (const :tag "EAF PDF Viewer" eaf-pdf-viewer)
+	  (const :tag "EAF System Monitor" eaf-system-monitor)
+	  (const :tag "EAF Terminal" eaf-terminal)
+	  (const :tag "EAF Video Player" eaf-video-player)
+	  (const :tag "EAF Vue Demo" eaf-vue-demo)
+	  (const :tag "EAF NetEase Cloud Music" eaf-netease-cloud-music)
+	  (const :tag "EAF RSS Reader" eaf-rss-reader)
+))
```

Comment:
Please simply read the `applications.json` to retrieve the available applications. The EAF code shouldn't explicitly list the apps.

Reply:
I agree it would be much better, but I have no idea how to do it. Do you have experience with json in emacs?

As it stands, the README also repeats the list of apps.

Reply:
> As it stands, the README also repeats the list of apps.

You're right, we missed that part. :)

> I agree it would be much better, but I have no idea how to do it. Do you have experience with json in emacs?

Parsing json isn't too hard in Emacs, I recommend you to have a look at [this part](https://www.gnu.org/software/emacs/manual/html_node/elisp/Parsing-JSON.html). Let me know if you need some help.

Reply:
Okay, thanks for the pointer. 

I was also scared of making the whole thing work with defcustom, but it turns out that passing the output of a suitable function to the :type keyword works, no need to fiddle with macros.